run.py

```diff
@@ -12,25 +12,28 @@
 from utils import report_generator
 
 
-def main(gear_context):
+def main(fw, gear_context):
     """Orchestrate dicom-send gear."""
     log.info("Starting dicom-send gear.")
 
     # Prepare gear arguments by parsing the gear configuration
-    gear_args, download = parse_config.generate_gear_args(gear_context)
+    gear_args, session_id, download, input_dir, api_key =\
+        parse_config.generate_gear_args(fw, gear_context)
```

Comment:
This function should pass back the necessary parameters and have fw inside the parse_config script. In fact, that is already what you have. But instead of fw in run.py, import fw inside parse_config.py and remove all other fw mentions in run.py - they are not needed.